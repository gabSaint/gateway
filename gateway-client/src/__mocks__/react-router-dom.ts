import React, { FunctionComponent } from "react";

const useParams = jest.fn();
const useHistory = jest.fn();

const Link: FunctionComponent = ({ children }) =>
  React.createElement("div", null, children);

export { useParams, useHistory, Link };
