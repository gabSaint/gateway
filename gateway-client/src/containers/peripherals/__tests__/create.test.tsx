import React from "react";
import { ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import { useParams } from "react-router-dom";
import renderer from "react-test-renderer";

import CreatePeripheral from "../create";

describe("CreatePeripheral", () => {
  let wrapperCreatePeripheral: ReactWrapper | ShallowWrapper;

  beforeEach(() => {
    // @ts-ignore
    useParams.mockImplementationOnce(() => ({ id: 1 }));
    wrapperCreatePeripheral = shallow(<CreatePeripheral />);
  });

  it("renders correctly", () => {
    // @ts-ignore
    useParams.mockImplementationOnce(() => ({ id: "1" }));

    const tree = renderer.create(<CreatePeripheral />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders title", () => {
    const title = wrapperCreatePeripheral.find("h2");
    expect(title).toHaveLength(1);
  });

  it("renders form", () => {
    const form = wrapperCreatePeripheral.find("PeripheralForm");
    expect(form).toHaveLength(1);
  });
});
