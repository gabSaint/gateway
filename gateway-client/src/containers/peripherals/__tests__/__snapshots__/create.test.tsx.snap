// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePeripheral renders correctly 1`] = `
Array [
  <h2>
    Create Peripheral
  </h2>,
  <form
    action="#"
    className="ui form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="input-vendor"
      >
        Vendor
      </label>
      <input
        id="input-vendor"
        name="vendor"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="inline fields"
      role="group"
    >
      <label>
        Status
      </label>
      <div
        className="field"
      >
        <div
          className="ui radio checkbox"
        >
          <input
            checked={true}
            className="radio-item"
            name="status"
            onBlur={[Function]}
            onChange={[Function]}
            tabIndex="0"
            type="radio"
            value="online"
          />
          <label>
            Online
          </label>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui radio checkbox"
        >
          <input
            checked={false}
            className="radio-item"
            name="status"
            onBlur={[Function]}
            onChange={[Function]}
            tabIndex="0"
            type="radio"
            value="offline"
          />
          <label>
            Offline
          </label>
        </div>
      </div>
    </div>
    <button
      data-test="button-submit"
      type="submit"
    >
      Submit
    </button>
    <div>
      <button
        data-test="button-cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>,
]
`;
